Automattic/wp-calypso
# Make sure failing builds fail

## Proposed Changes

When a webpack build error occurs, the CI system will not stop. This is because the failure gets swallowed up like so:

```sh
RUN yarn run build 2>&1 | tee /tmp/build_log.txt
```

The purpose of this line is to echo all output to CI (including errors), while also saving it to a file so that another step can check it for different warnings. However, if `yarn run build` has a non-0 code, `tee` will still run, and will still have a 0 exit code. So process does not exit, and CI continues. ([Here's an example in CI](https://teamcity.a8c.com/buildConfiguration/calypso_BuildDockerImage/9952769?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&showLog=9952769_663_168&logView=flowAware))

This results in a broken Docker image -- not containing a valid Calypso build -- being deployed to staging, which breaks Calypso for internal users. 

To solve this problem, we have to set `pipefail` to ensure a failure in the first part of the pipe fails the entire pipe. We do this by setting the `SHELL` directive in our Dockerfiles.

## Testing Instructions
- I've pushed a temporary commit which causes a failure so we can observe if the solution works.
- I'll also test the Slack integration and add more helpful alerts.
